drivers/pci/pcie_brcmstb.c

```diff
@@ -186,6 +246,52 @@ static bool brcm_pcie_rc_mode(struct brcm_pcie *pcie)
 	return (val & STATUS_PCIE_PORT_MASK) >> STATUS_PCIE_PORT_SHIFT;
 }
 
+static bool brcm_pcie_rc_mode_2712(struct brcm_pcie *pcie)
+{
+	void __iomem *base = pcie->base;
+	u32 val = readl(base + PCIE_MISC_PCIE_STATUS);
+
+	return !!FIELD_GET(STATUS_PCIE_PORT_MASK_2712, val) | 1;
```

Comment:
This worth investigation and some comment at least.

Reply:
Switched to brcm_pcie_rc_mode and remove special implementation for 2712. looks like it is a mistake because it is not likely that there it unlikely that port mask was changed in the same controller. That's why on the linux code this hack was applied.